actions/setup-nix-gati/action.yml

```diff
@@ -0,0 +1,77 @@
+name: setup-nix-gati
+description: "Setup the environment for Nix using GATI"
+
+inputs:
+  # general inputs
+  aws-role-arn:
+    description: ARN of role capable of getting token from GATI
+    required: true
+  aws-lambda-url:
+    description: URL of GATI lambda function
+    required: true
+  aws-region:
+    description: AWS region
+    required: true
+  aws-role-duration-seconds:
+    description: Duration of role in seconds
+    required: false
+    default: "900"
+  enable-magic-cache:
+    description: "Enable magic nix cache"
+    required: false
+    default: false
+  # grafana inputs
+  metrics-job-name:
+    description: "grafana metrics job name"
+    required: false
+    default: setup-nix-gati
+  gc-host:
+    description: "grafana hostname"
+    required: false
+  gc-basic-auth:
+    description: "grafana basic auth"
+    required: false
+  gc-org-id:
+    description: "grafana org/tenant id"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Setup GitHub token
+      id: setup-github-token
+      uses: smartcontractkit/.github/actions/setup-github-token@main
+      with:
+        aws-role-arn: ${{ inputs.aws-role-arn }}
+        aws-lambda-url: ${{ inputs.aws-lambda-url }}
+        aws-region: ${{ inputs.aws-region }}
+        aws-role-duration-seconds: ${{ inputs.aws-role-duration-seconds }}
+
+    - name: Configure GitHub token from GATI
+      shell: bash
+      run: |
+        # redirects ssh url to https with the token
+        git config --global \
+          url."https://x-access-token:${{ steps.setup-github-token.outputs.access-token }}@github.com/smartcontractkit/".insteadOf \
+          "ssh://git@github.com/smartcontractkit/"
+
+    - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c #v26
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+        extra_nix_config: |
+          sandbox = true # force sandbox for all OS (normally disabled for macOS)
+
+    - name: use magic nix cache
+      if: inputs.enable-magic-cache == 'true'
+      uses: DeterminateSystems/magic-nix-cache-action@87e8236f46702ab0ce5a058b605a173ec88d618e #v6
+
+    - name: Collect metrics
+      if: always()
```

Comment:
This should be conditional on `metrics-job-name` being passed.

See this as an example diff: https://github.com/smartcontractkit/.github/pull/421/files

Reply:
I can make the fix here, and I think we also need to update the action generation to have this fix.

Reply:
Yes good catch - I will do that.